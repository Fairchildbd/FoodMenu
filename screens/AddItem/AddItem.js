import React, { Fragment, useState, useContext } from 'react';
import { View, TouchableOpacity, Text } from 'react-native';
import { Header, Icon, Input, Button } from 'react-native-elements';
import { headerStyles } from '../../styles/styles';

export function AddItem(props) {
  const [img, setImage] = useState('');
  const [name, setName] = useState('');
  const [description, setDescription] = useState('');
  const [price, setPrice] = useState('');
  const payload = {
    img,
    name,
    description,
    price
  };
  return (
    <Fragment>
      <Header
        backgroundColor='#f24e49'
        leftComponent={
          <View style={headerStyles.headerLeft}>
            <TouchableOpacity style={ headerStyles.headerButtonContainer } onPress={() => props.navigation({ route: 'Home' })}>
              <Icon name="arrow-back-ios" color="white" />
              <Text style={ headerStyles.headerButtonText }>Go Back</Text>
            </TouchableOpacity>
          </View>
        }
        centerComponent={ {text: 'Add Item', style: headerStyles.heading} }
      />
      <View style={ { margin: 10 } }>
        <Input
          placeholder='Image URL'
          leftIcon={{ type: 'font-awesome', name: 'chevron-left' }}
          onChangeText={ value => setImage(value) }
        />
        <Input
          placeholder='Title of Item'
          leftIcon={{ type: 'font-awesome', name: 'chevron-left' }}
          onChangeText={ value => setName(value) }
        />
        <Input
          placeholder='Description of Item'
          leftIcon={{ type: 'font-awesome', name: 'chevron-left' }}
          onChangeText={ value => setDescription(value) }
        />
        <Input
          placeholder='Price of Item'
          leftIcon={{ type: 'font-awesome', name: 'chevron-left' }}
          onChangeText={ value => setPrice(value) }
        />
        <Button testID='Submission' buttonStyle={ { backgroundColor: '#5bd1d7' } } onPress={ () => props.navigation({ route: 'Home', params: payload }) } title='Submit' />
      </View>
    </Fragment>
  )
}
