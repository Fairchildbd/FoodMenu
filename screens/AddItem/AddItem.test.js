import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent, cleanup } from '@testing-library/react-native';
import { AddItem } from './AddItem';

describe('<AddItem />', () => {
  afterAll(cleanup);
  const payload = {
    img: 'https://api.pizzahut.io/v1/content/en-in/in-1/images/pizza/margherita.7d3b623ceb66e87fc7247fa81c6cfdc1.1.jpg?width=800',
    name: 'Cheese Pizza',
    description: 'It has a lot of cheese',
    price: '$10.25'
  };
  const props = {
    navigation: jest.fn()
  };
  it('can render the screen', () => {
    const tree = renderer.create(<AddItem { ...props }/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('can fire a submission', () => {
    const { getByTestId, getByPlaceholderText } = render(<AddItem { ...props } />);
    fireEvent.changeText(getByPlaceholderText('Image URL'), payload.img);
    fireEvent.changeText(getByPlaceholderText('Title of Item'), payload.name);
    fireEvent.changeText(getByPlaceholderText('Description of Item'), payload.description);
    fireEvent.changeText(getByPlaceholderText('Price of Item'), payload.price);
    fireEvent.press(getByTestId('Submission'), { route: 'Home', params: payload });
    expect(props.navigation).toHaveBeenCalledWith({ route: 'Home', params: payload });
  });
  it('can go back home with no params', () => {
    const { getByText } = render(<AddItem { ...props } />);
    fireEvent.press(getByText('Go Back'), { route: 'Home'});
    expect(props.navigation).toHaveBeenCalledWith({ route: 'Home'});
  });
});
