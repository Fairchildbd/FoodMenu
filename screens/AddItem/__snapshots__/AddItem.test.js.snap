// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddItem /> can render the screen 1`] = `
Array [
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#f24e49",
        "borderBottomColor": "#f2f2f2",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 10,
        "paddingVertical": 10,
      }
    }
    testID="headerContainer"
  >
    <RNCSafeAreaView
      edges={
        Array [
          "left",
          "top",
          "right",
        ]
      }
      style={
        Object {
          "flexDirection": "row",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "marginLeft": 10,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "overflow": "hidden",
                }
              }
            >
              <View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text />
                </View>
              </View>
            </View>
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontSize": 16,
                }
              }
            >
              Go Back
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 3,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "#fff",
              "fontSize": 22,
              "fontWeight": "bold",
            }
          }
          text="Add Item"
        >
          Add Item
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "flex": 1,
          }
        }
      />
    </RNCSafeAreaView>
  </View>,
  <View
    style={
      Object {
        "margin": 10,
      }
    }
  >
    <View
      style={
        Object {
          "paddingHorizontal": 10,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#86939e",
            "flexDirection": "row",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "height": 40,
              "justifyContent": "center",
              "marginVertical": 4,
              "paddingRight": 4,
            }
          }
        >
          <View
            style={
              Object {
                "overflow": "hidden",
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
              >
                <Text />
              </View>
            </View>
          </View>
        </View>
        <TextInput
          allowFontScaling={true}
          editable={true}
          onChangeText={[Function]}
          placeholder="Image URL"
          placeholderTextColor="#86939e"
          rejectResponderTermination={true}
          style={
            Object {
              "color": "#242424",
              "flex": 1,
              "fontSize": 18,
              "minHeight": 40,
            }
          }
          testID="RNE__Input__text-input"
          underlineColorAndroid="transparent"
        />
      </View>
      <Text
        style={
          Object {
            "color": "#ff190c",
            "fontSize": 12,
            "margin": 5,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "paddingHorizontal": 10,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#86939e",
            "flexDirection": "row",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "height": 40,
              "justifyContent": "center",
              "marginVertical": 4,
              "paddingRight": 4,
            }
          }
        >
          <View
            style={
              Object {
                "overflow": "hidden",
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
              >
                <Text />
              </View>
            </View>
          </View>
        </View>
        <TextInput
          allowFontScaling={true}
          editable={true}
          onChangeText={[Function]}
          placeholder="Title of Item"
          placeholderTextColor="#86939e"
          rejectResponderTermination={true}
          style={
            Object {
              "color": "#242424",
              "flex": 1,
              "fontSize": 18,
              "minHeight": 40,
            }
          }
          testID="RNE__Input__text-input"
          underlineColorAndroid="transparent"
        />
      </View>
      <Text
        style={
          Object {
            "color": "#ff190c",
            "fontSize": 12,
            "margin": 5,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "paddingHorizontal": 10,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#86939e",
            "flexDirection": "row",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "height": 40,
              "justifyContent": "center",
              "marginVertical": 4,
              "paddingRight": 4,
            }
          }
        >
          <View
            style={
              Object {
                "overflow": "hidden",
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
              >
                <Text />
              </View>
            </View>
          </View>
        </View>
        <TextInput
          allowFontScaling={true}
          editable={true}
          onChangeText={[Function]}
          placeholder="Description of Item"
          placeholderTextColor="#86939e"
          rejectResponderTermination={true}
          style={
            Object {
              "color": "#242424",
              "flex": 1,
              "fontSize": 18,
              "minHeight": 40,
            }
          }
          testID="RNE__Input__text-input"
          underlineColorAndroid="transparent"
        />
      </View>
      <Text
        style={
          Object {
            "color": "#ff190c",
            "fontSize": 12,
            "margin": 5,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "paddingHorizontal": 10,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#86939e",
            "flexDirection": "row",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "height": 40,
              "justifyContent": "center",
              "marginVertical": 4,
              "paddingRight": 4,
            }
          }
        >
          <View
            style={
              Object {
                "overflow": "hidden",
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
              >
                <Text />
              </View>
            </View>
          </View>
        </View>
        <TextInput
          allowFontScaling={true}
          editable={true}
          onChangeText={[Function]}
          placeholder="Price of Item"
          placeholderTextColor="#86939e"
          rejectResponderTermination={true}
          style={
            Object {
              "color": "#242424",
              "flex": 1,
              "fontSize": 18,
              "minHeight": 40,
            }
          }
          testID="RNE__Input__text-input"
          underlineColorAndroid="transparent"
        />
      </View>
      <Text
        style={
          Object {
            "color": "#ff190c",
            "fontSize": 12,
            "margin": 5,
          }
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "borderRadius": 3,
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 3,
          },
          undefined,
          false,
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="Submission"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#5bd1d7",
              "borderColor": "#2089dc",
              "borderRadius": 3,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingVertical": 1,
                "textAlign": "center",
              }
            }
          >
            Submit
          </Text>
        </View>
      </View>
    </View>
  </View>,
]
`;
